### DISCUSSION TITLE
Cannot read properties of null (reading 'gasLimit')

### DISCUSSION THREAD

**Discussion by johnpaulcas** [wevm/viem#679] [Question]
When we try to interact on the polygon mainnet, we are having an issue saying "cannot read properties of null (reading 'gasLimit')" By the way, we are not passing this data on our code, we assume the library automatically calculates this.

<img width="1271" alt="Screenshot 2023-06-09 at 9 52 18 AM" src="https://github.com/wagmi-dev/viem/assets/44362950/c43b6938-fa4c-4dbb-a482-92089a0a4ce5">

**Comment by jxom** [comment]
This seems to be a MetaMask error – how exactly are you getting this error? can you provide some code?

**Reply by HenryDotApp** [reply]
Hi @jxom, also running into this error while using Viem with Polygon, but it seems to happen very flakily and randomly, and I can't reliably reproduce it

For my context, `sendTransaction` getting called here: https://github.com/reservoirprotocol/reservoir-kit/blob/main/packages/sdk/src/utils/viemWallet.ts#L69

I see in Viem's repo that `.gasLimit` is called here: https://github.com/wagmi-dev/viem/blob/339fbf1323fb38c1c5b4ab7388577327f8aee682/src/utils/formatters/block.ts#L54 but it doesn't make sense why that line would throw, and not an earlier line, if `block` is null

Please let me know what other info I can provide that would help narrow this down!

**Reply by cryptoflop** [reply]
I also got this error with the `publicProvider` but was able to resolve it with the ankr rpc (maybe others work too).
But even with a custom rpc endpoint the gas estimation didn't work and created a tx with 0 gas...
Really clueless if it's a rpc, metamask, or wagmi/viem problem here...

**Reply by karacurt** [reply]
Hey @johnpaulcas, @Henry-MagicEden, @cryptoflop ! i am also getting an error similar to this one! Let me know if anyone was able to solve it!

**Reply by BravoNatalie** [reply]
try to use the `gas: bigint` parameter for the gas limit. 
https://viem.sh/docs/accounts/signTransaction#gas-optional
